pallets/assets-registry/src/lib.rs

```diff
@@ -183,6 +183,94 @@ pub mod pallet {
 		}
 	}
 
+	// Should adapter the representation of asset location after reanchored.
+	// Because xcm would reanchore the location of the asset that reserved on our chain.
+	// https://github.com/paritytech/polkadot/pull/4470
+	pub trait ReserveAssetChecker {
+		// Return true if asset is reserved on local
+		fn is_reserve_asset(asset: &MultiAsset) -> bool;
+		// Return true if given location is reserved on local
+		fn is_reserve_asset_location(id: &MultiLocation) -> bool;
+		// Return asset origin location if asset is reserved on local, otherwise return None
+		fn to_absoluted_location(location: &MultiLocation) -> Option<MultiLocation>;
```

Comment:
Syntax: `to_absolute_location`.

Why do you call this function like this? It sounds like you want something like `to_long_location_only_local`.

Reply:
Yes, I was missing mind, let's rename it to `to_gloableconsensus_location`.